dash/include/dash/GlobRef.h

```diff
@@ -335,24 +338,260 @@ class GlobRef
+    return GlobPtr<atomic_t>(_gptr);
+  }
+  
+  dart_gptr_t dart_gptr() const {
+    return _gptr;
+  }
+  
+  /**
+   * Checks whether the globally referenced element is in
+   * the calling unit's local memory.
+   */
+  bool is_local() const {
+    return GlobPtr<T>(_gptr).is_local();
+  }
+  
+  /// atomically assigns value
+  GlobRef<atomic_t> operator=(const T & value) const {
+    dash::atomic::store(*this, value);
+    return *this;
+  }
+  
+  void store(const T & value) const {
+    dash::atomic::store(*this, value);
+  }
+  /// atomically fetches value
+  T load() const {
+    return dash::atomic::load(*this);
+  }
+  
+  /**
+   * atomically exchanges value
+   */
+  T exchange(const T & value) const {
+    return dash::atomic::exchange(*this, value);
+  }
+  
+  /**
+   * Atomically compares the value with the value of expected and if those are
+   * bitwise-equal, replaces the former with desired.
+   * 
+   * \note not implemented yet
+   * 
+   * \return  True if value is exchanged
+   * 
+   * \see \c dash::atomic::compare_exchange
+   */
+  bool compare_exchange(const T & expected, const T & desired) const {
+    return dash::atomic::compare_exchange(*this, expected, desired);
+  }
+  
+  /*
+   * ---------------------------------------------------------------------------
+   * ------------ specializations for atomic integral types --------------------
+   * ---------------------------------------------------------------------------
+   *
+   *  As the check for integral type is already implemented in the 
+   *  dash::atomic::* algorithms, no check is performed here
+   */
+  
+  /**
+   * DASH specific variant which is faster than \cfetch_add
+   * but does not return value
+   */
+  void add(const T & value) const {
+    dash::atomic::add(*this, value);
+  }
+  
+  T fetch_add(const T & value) const {
+    return dash::atomic::fetch_add(*this, value);
+  }
+  
+  /**
+   * DASH specific variant which is faster than \cfetch_sub
+   * but does not return value
+   */
+  void sub(const T & value) const {
+    dash::atomic::sub(*this, value);
+  }
+  
+  T fetch_sub(const T & value) const {
+    return dash::atomic::fetch_sub(*this, value);
+  }
+  
+  /// prefix atomically increment value by one
+  T operator++ () const {
+    return fetch_add(1) + 1;
+  }
+  
+  /// postfix atomically increment value by one
+  T operator++ (int) const {
+    return fetch_add(1);
+  }
+  
+  /// prefix atomically decrement value by one
+  T operator-- () const {
+    return fetch_sub(1) - 1;
+  }
+  
+  /// postfix atomically decrement value by one
+  T operator-- (int) const {
+    return fetch_sub(1);
+  }
+  
+  /// atomically increment value by ref
+  T operator+=(const T & value) const {
+    return fetch_add(value) + value;
+  }
+  
+  /// atomically decrement value by ref
+  T operator-=(const T & value) const {
+    return fetch_sub(value) - value;
+  }
+  
+private:
+  
+  dart_gptr_t _gptr;
+  
+};
+
+
```

Comment:
I suggest moving this specialization to `Atomic.h` as non-atomic use does not require it to compile successfully.